mesonbuild/dependencies/misc.py

```diff
@@ -32,6 +32,42 @@
 )
 
 
+class CoarrayDependency(ExternalDependency):
+    """
+    Coarrays are a Fortran 2008 feature.
+    Coarrays are sometimes implemented via external library (GCC+OpenCoarrays),
+    while other compilers just build in support (Cray, IBM, Intel, NAG).
+    Coarrays may be thought of as a high-level language abstraction of
+    low-level MPI calls.
+    """
+    def __init__(self, environment, kwargs):
+        super().__init__('coarray', environment, 'fortran', kwargs)
+        kwargs['required'] = False
```

Comment:
Are there actually any kwargs from the parent call that you need to push down to the CMake call, or can you just craft a new kwargs dict in the GCC path?

Reply:
It's possible that the user wants to pass down through `cmake_module_path` in kwargs, for non-system installs of OpenCoarrays as one example